n2y/main.py

```diff
@@ -61,67 +65,83 @@ def name_column_valid(raw_rows, name_column):
     return True
 
 
 def export_markdown(client, raw_rows, options):
     file_names = []
+    skips = {'Empty': 0, 'Unnamed': 0, 'Duplicate': 0}
     for row in raw_rows:
         meta = simplify.flatten_database_row(row)
         page_name = meta[options.name_column]
-        filename = re.sub(r"[/,\\]", '_', page_name.lower())
         if page_name is not None:
-            print(f"Processing {meta[options.name_column]}", file=sys.stderr)
-            if filename in file_names:
-                print(f'WARNING: duplicate file name "{filename}.md"', file=sys.stderr)
-            file_names.append(filename)
-
-            pandoc_output = converter.load_block(client, row['id']).to_pandoc()
-            # do not create markdown pages if there is no page in Notion
-            if pandoc_output:
-                markdown = pandoc.write(pandoc_output, format='gfm+tex_math_dollars') \
-                    .replace('\r\n', '\n')  # Deal with Windows line endings
-
-                # create target path if it doesn't exist
-                # if not os.path.exists(options.target):
-                os.makedirs(options.target, exist_ok=True)
-
-                # sanitize file name just a bit
-                # maybe use python-slugify in the future?
-                with open(os.path.join(options.target, f"{filename}.md"), 'w') as f:
-                    f.write('---\n')
-                    f.write(yaml.dump(meta))
-                    f.write('---\n\n')
-                    f.write(markdown)
+            filename = re.sub(r"[\s/,\\]", '_', page_name.lower())
+            if filename not in file_names:
+                file_names.append(filename)
+
+                pandoc_output = converter.load_block(client, row['id']).to_pandoc()
+                # do not create markdown pages if there is no page in Notion
+                if pandoc_output:
+                    logger.warning(f'Processing Page "{meta[options.name_column]}"')
+                    markdown = pandoc.write(pandoc_output, format='gfm+tex_math_dollars') \
+                        .replace('\r\n', '\n')  # Deal with Windows line endings
+
+                    # create target path if it doesn't exist
+                    # if not os.path.exists(options.target):
+                    os.makedirs(options.target, exist_ok=True)
+
+                    # sanitize file name just a bit
+                    # maybe use python-slugify in the future?
+                    with open(os.path.join(options.target, f"{filename}.md"), 'w') as f:
+                        f.write('---\n')
+                        f.write(yaml.dump(meta))
+                        f.write('---\n\n')
+                        f.write(markdown)
+                else:
+                    logger.debug(f"Skipping Empty Page: {page_name}")
+                    skips['Empty'] += 1
             else:
-                print(f"WARNING: skipping empty page: {page_name}", file=sys.stderr)
+                logger.debug(
```

Comment:
This should be `error` or `warning`.

Reply:
I need it to be a level that less than the logger's level unless you want the "Processing" and "files skipped"  statements upped to a higher level than warning

Reply:
I'm not sure that I fully understand what you mean. I'm going off of this table

![image](https://user-images.githubusercontent.com/4360601/161874802-dc3f0548-3a4b-45c5-8eda-faa5111f81e9.png)
https://docs.python.org/3/howto/logging.html#logging-advanced-tutorial

Reply:
Because line 113 in main.py (an `info()` log) has to print to stdout, the level of the logger has been lowered to info. Now, since the whole point of making these changes is to make sure logs pertaining to skipping files _don't_ print to stdout, that means they have to be categorized to a logger level lower than info (i.e. debug).

Reply:
Here is what David asked for (issue #13):
> Finally, we should either remove the warning associated with blank files (since this is now expected behavior and there may be many rows with blank slugs) OR we should condense the warnings into a single line of text that says something like "Ignored one or more rows with blank slug column values."

I am not totally sure why a message telling the user to rename a file is a `debug` level message. (I read these particular lines as dealing with duplicates and not blanks)

Reply:
While David and I were discussing how to handle duplicate file names, we basically came to the conclusion that we should just skip files that were duplicated in favor of letting the end-users rename them independently instead of using the algorithm in the [dupefileWarning](https://github.com/innolitics/n2y/blob/dupefileWarning/n2y/main.py#:~:text=if%20filename%20in,file_names.append(filename) branch to rename them. Because I figured he would want those warnings added to that condensed final message, not printed immediately, I asked if that were true. He confirmed that my assumption was correct, so I added that functionality here.

Reply:
I'll defer to @johndgiese

Reply:
Reason I choose the debug level over the error level was the same reasoning as above.

Reply:
I think this message should be a warning, since we don't typically expect that two pages with the same name will be present. It doesn't feel like it's an error either though, since the software can continue running as expected.